src/validation.cairo

```diff
@@ -38,11 +39,76 @@ impl BlockValidatorImpl of BlockValidator {
 }
 
 #[generate_trait]
-impl TransactionValidatorImpl of TransactionValidator {
+pub impl TransactionValidatorImpl of TransactionValidator {
+    // marker, flag, and witness fields in segwit transactions are not included
+    // this means txid computation is the same for legacy and segwit tx
     fn txid(self: @Transaction) -> u256 {
-        // TODO: implement
-        0
+        // append version (1 byte)
+        let mut sha256_input: ByteArray = "";
+        sha256_input.append_word((*self.version).into(), 1);
+
+        // append padding (3 bytes)
+        sha256_input.append_word(0, 3);
+
+        // append inputs count (1 byte)
+        sha256_input.append_word((*self.inputs).len().into(), 1);
```

Comment:
Note that this should actually be compact size encoding which can be larger than 1 byte for larger number of inputs (same applies to outputs)

https://learnmeabitcoin.com/technical/general/compact-size/

Reply:
does it also apply to  ScriptSig size  ? Can ScriptSig be big enough so that ScriptSig size doesnt fit into 1 byte?